src/diff_edit.rs

```diff
@@ -96,6 +135,122 @@ fn set_readonly_recursively(path: &Path) -> Result<(), std::io::Error> {
     }
 }
 
+pub fn run_mergetool(
+    _ui: &mut Ui,
+    _settings: &UserSettings,
+    tree: &Tree,
+    repo_path: &RepoPath,
+) -> Result<TreeId, DiffEditError> {
+    let conflict_id = match tree.path_value(repo_path) {
+        Some(TreeValue::Conflict(id)) => id,
+        Some(_) => return Err(DiffEditError::NotAConflictError(repo_path.clone())),
+        None => return Err(DiffEditError::PathNotFoundError(repo_path.clone())),
+    };
+    let conflict = tree.store().read_conflict(repo_path, &conflict_id)?;
+    let mut materialized_conflict: Vec<u8> = vec![];
+    materialize_conflict(
+        tree.store(),
+        repo_path,
+        &conflict,
+        &mut materialized_conflict,
+    )
+    .unwrap_or_else(|_| materialized_conflict.clear());
+    // For normal files, this is the default user-facing represenation of conflicts
+    // as a text file with conflict markers.
+    let materialized_conflict = materialized_conflict;
+
+    let content = match extract_file_conflict_data(tree.store(), repo_path, &conflict) {
+        Some(c) => c,
+        _ => {
+            // In this case, `materialized_conflict` is a short summary of what kind of
+            // files are involved in the conflict.
+            let summary = match std::str::from_utf8(materialized_conflict.as_slice()) {
+                Ok(s) if !materialized_conflict.is_empty() => s.to_string(),
+                _ => "<Failed to create a readable summary of the conflict>".to_string(),
+            };
+            return Err(DiffEditError::NotNormalFilesError(
+                repo_path.clone(),
+                summary,
+            ));
+        }
+    };
+    if content.removes.len() > 1 || content.adds.len() > 2 {
+        return Err(DiffEditError::ConflictTooComplicatedError {
+            path: repo_path.clone(),
+            removes: content.removes.len(),
+            adds: content.adds.len(),
+        });
+    };
+
+    let temp_dir = tempfile::Builder::new()
+        .prefix("jj-merge-")
+        .tempdir()
+        .map_err(DiffEditError::SetUpDirError)?;
+    let files_maybe = [
+        ("base", content.removes.get(0)),
+        ("left", content.adds.get(0)),
+        ("right", content.adds.get(1)),
+        ("output", Some(&materialized_conflict)),
+    ];
+    let files: HashMap<&str, _> = files_maybe
+        .map(|(name, f)| (name, f.cloned().unwrap_or_default()))
+        .iter()
+        .cloned()
+        .collect();
+    let paths: Result<HashMap<&str, _>, DiffEditError> = files
+        .iter()
+        .map(|(name, contents)| {
+            let path = temp_dir.path().join(name);
```

Comment:
It's probably better to use the original file name/suffix. Merge tool can enable syntax highlight based on filename.

e.g. `(base|left|right|output)/{basename}`, `{basename_wo_suffix}~(base|left|right|output).{suffix}`

Reply:
This is a good point. I'll probably go with `{base|left|right|output}_{basename_with_suffix}`. In the editor screen, I want it to be clear which file is which.